sample-app/src/components/VisualAutocomplete/VisualSearchBar.tsx

```diff
@@ -50,10 +57,27 @@ export default function VisualSearchBar({
   const answersActions = useAnswersActions();
   const query = useAnswersState(state => state.query.input) ?? '';
   const isLoading = useAnswersState(state => state.searchStatus.isLoading) ?? false;
-  const [executeQuery, autocompletePromiseRef] = useSearchWithNearMeHandling(answersActions)
+  const [executeQueryWithNearMeHandling, autocompletePromiseRef] = useSearchWithNearMeHandling(answersActions)
   const [autocompleteResponse, executeAutocomplete] = useSynchronizedRequest(async () => {
     return answersActions.executeUniversalAutocomplete();
   });
+  const recentSearches = useMemo(() => {
+    if (hideRecentSearches) {
+      localStorage.removeItem(DEFAULT_RECENT_SEARCH_KEY);
+      return undefined;
+    } else {
+      return new RecentSearches({ limit: recentSearchesLimit, namespace: DEFAULT_RECENT_SEARCH_KEY });
+    }
+  }, [recentSearchesLimit, hideRecentSearches]);
+  const haveRecentSearches = !hideRecentSearches && recentSearches?.getRecentSearches() && recentSearches.getRecentSearches().length > 0;
```

Comment:
does this work? just removing the last &&
```ts
const haveRecentSearches = !hideRecentSearches && recentSearches?.getRecentSearches()?.length > 0
```

Reply:
nope, attempted that before and it error out for comparing undefined with a number

Reply:
maybe `length !== 0` instead of `> 0`? alternatively I wouldn't mind breaking this up into two lines